src/solve.jl

```diff
@@ -100,62 +113,78 @@ function _solve(cp::ControlledPlant,
         "the dimension of the initial states of the initial-value problem to " *
         "be $(n + m + q), but it is $(dim(Q₀))"))
 
-    if m > 0
-        W₀ = project(Q₀, in_vars)
-    end
+    W₀ = m > 0 ? project(Q₀, in_vars) : nothing
 
-    ti = tstart(time_span)
-    Δti = zero(ti)
-    NSAMPLES = ceil(Int, diam(time_span) / sampling_time)
+    function nnet_forward(X)
+        X = apply(preprocessing, X)
+        U = forward_network(solver, network, X)
+        U = apply(normalization, U)
+        if dim(U) == 1  # simplify the control input for intervals
+            U = overapproximate(U, Interval)
+        end
+        return U
+    end
 
-    # preallocate output flowpipe
+    # preallocate output flowpipes
     sol = nothing
     NT = numtype(cpost)
     RT = rsetrep(cpost)
     FT = Flowpipe{NT, RT, Vector{RT}}
-    out = Vector{FT}(undef, NSAMPLES)
+    flowpipes = Vector{FT}()
+    controls = Vector()
+
+    # waiting list
+    waiting_list = Vector{ControlFlowpipe{FT}}()
 
-    for i in 1:NSAMPLES
-        if i == 1
+    # iterate
+    first = true
+    while true
+        if first
             X = nothing
             X₀ = project(Q₀, st_vars)
+            k = 1
+            first = false
+        elseif isempty(waiting_list)
```

Comment:
for clarity may we move this check to the main while? `while !isempty(waiting_list) # something to process...` i would go further end have the initialization step outside the loop but that's a matter of taste :smile:

Reply:
I wanted to avoid code duplication. The body below is the same for the first case and the other cases. I can pull it out.

Reply:
Done :+1: